simplified-tests/src/test/java/org/nypl/simplified/tests/opds2/OPDS2ParserContract.kt

```diff
@@ -0,0 +1,53 @@
+package org.nypl.simplified.tests.opds2
+
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.nypl.simplified.opds2.parser.api.OPDS2ParsersType
+import org.nypl.simplified.parser.api.ParseResult
+import org.slf4j.Logger
+import java.io.FileNotFoundException
+import java.io.InputStream
+import java.net.URI
+
+abstract class OPDS2ParserContract {
```

Comment:
This is interesting. I used to use parametrized tests in such cases, is there any specific rationale behind your choice?